server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroupMetadata.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.Diff;
+import org.opensearch.cluster.DiffableUtils;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.opensearch.cluster.metadata.Metadata.ALL_CONTEXTS;
+
+/**
+ * This class holds the resourceLimitGroupMetadata
+ * sample schema
+ * {
+ *     "resourceLimitGroups": {
+ *         "name": {
+ *             {@link ResourceLimitGroup}
+ *         },
+ *        ...
+ *     }
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroupMetadata implements Metadata.Custom {
+    public static final String TYPE = "resourceLimitGroup";
+    private static final ParseField RESOURCE_LIMIT_GROUP_FIELD = new ParseField("resourceLimitGroups");
```

Comment:
`"resourceLimitGroup"` ?

Reply:
The metadata will contain all the ResourceLimitGroups in the cluster.